lib/SILGen/SILGen.cpp

```diff
@@ -732,46 +732,37 @@ void SILGenModule::emitAbstractFuncDecl(AbstractFunctionDecl *AFD) {
-      // spcified.
-      StringRef primName, adjName, jvpName, vjpName;
-      bool hasPrimitiveAdjoint = false;
-      if (auto *primFn = diffAttr->getPrimalFunction())
-        primName = getFunction(SILDeclRef(primFn), ForDefinition)->getName();
-      if (auto *adjointFn = diffAttr->getAdjointFunction()) {
-        // If the adjoint is specified but the primal is not, then we treat the
-        // original as the primal.
-        if (primName.empty())
-          primName = silOriginalFn->getName();
-        adjName = getFunction(SILDeclRef(adjointFn), ForDefinition)->getName();
-        hasPrimitiveAdjoint = true;
-      }
-      else {
-        assert(primName.empty() &&
-               "Primal cannot be present if adjoint is not");
-      }
-      if (auto *jvpFn = diffAttr->getJVPFunction())
-        jvpName = getFunction(SILDeclRef(jvpFn), ForDefinition)->getName();
-      if (auto *vjpFn = diffAttr->getVJPFunction())
-        vjpName = getFunction(SILDeclRef(vjpFn), ForDefinition)->getName();
-      // Get lowered argument indices.
-      auto paramIndices = diffAttr->getCheckedParameterIndices()->getLowered(
-          AFD->getInterfaceType()->castTo<AnyFunctionType>());
-      SILAutoDiffIndices indices(/*source*/ 0, paramIndices);
-      silOriginalFn->addDifferentiableAttr(
-          SILDifferentiableAttr::create(
-            M, indices, primName, adjName,
-            /*primitive*/ hasPrimitiveAdjoint, jvpName, vjpName));
-      break;
+    auto silOriginalFn = getFunction(SILDeclRef(AFD), ForDefinition);
+    // Either only adjoint is specified, or both primal and adjoint are
+    // spcified.
+    StringRef primName, adjName, jvpName, vjpName;
+    bool hasPrimitiveAdjoint = false;
+    if (auto *primFn = diffAttr->getPrimalFunction())
+      primName = getFunction(SILDeclRef(primFn), ForDefinition)->getName();
+    if (auto *adjointFn = diffAttr->getAdjointFunction()) {
+      // If the adjoint is specified but the primal is not, then we treat the
+      // original as the primal.
+      if (primName.empty())
+        primName = silOriginalFn->getName();
+      adjName = getFunction(SILDeclRef(adjointFn), ForDefinition)->getName();
+      hasPrimitiveAdjoint = true;
     }
+    else {
+      assert(primName.empty() &&
+             "Primal cannot be present if adjoint is not");
     }
+    if (auto *jvpFn = diffAttr->getJVPFunction())
+      jvpName = getFunction(SILDeclRef(jvpFn), ForDefinition)->getName();
+    if (auto *vjpFn = diffAttr->getVJPFunction())
+      vjpName = getFunction(SILDeclRef(vjpFn), ForDefinition)->getName();
+    // Get lowered argument indices.
+    auto paramIndices = diffAttr->getCheckedParameterIndices()->getLowered(
+        AFD->getInterfaceType()->castTo<AnyFunctionType>());
+    SILAutoDiffIndices indices(/*source*/ 0, paramIndices);
+    silOriginalFn->addDifferentiableAttr(
+        SILDifferentiableAttr::create(
+          M, indices, primName, adjName,
```

Comment:
```suggestion
            M, indices, primName, adjName,
```

Reply:
Done in https://github.com/apple/swift/pull/21544/commits/adff0ab2125678960de1c27ba1d6eeede181b545